packages/diagrams/frontend/sirius-components-diagrams/src/renderer/DiagramRenderer.tsx

```diff
@@ -154,6 +154,12 @@ export const DiagramRenderer = ({ diagramRefreshedEventPayload }: DiagramRendere
     }
   }, [diagramRefreshedEventPayload, diagramDescription]);
 
+  useEffect(() => {
+    if (edges.length > 0) {
+      setEdges(edges.map((edge) => ({ ...edge, updatable: !!edge.selected })));
+    }
+  }, [JSON.stringify(edges)]);
```

Comment:
`JSON.stringify` does not make it clear why we rerender and will make us rerender with any change in any edges.
I think something like [selection.entries.map((selection) => selection.id).join()] or anything more "precise" will be more efficient.
Also `edges.length> 1` is doing nothing since if edges changes that would mean you have more than one edge

Reply:
done